docker-compose.yaml

```diff
@@ -0,0 +1,23 @@
+version: "3.5"
+services:
+
+  db:
+    image: mysql
+    restart: always
+    ports:
+      - "3308:3306"
+    environment:
+      MYSQL_ROOT_PASSWORD: 12345
+      MYSQL_USER: user
+      MYSQL_PASSWORD: 12345
+      MYSQL_DATABASE: books
```

Comment:
The recommended best practice is to externalize configuration and keep it out of the codebase. This means using environment variables, .env files, or configuration servers to manage different environments.

Your approach is less flexible since changes to the database connection details would require changes to the codebase and rebuilding the Docker image.